pkg/snmp/traps/config.go

```diff
@@ -0,0 +1,61 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2020 Datadog, Inc.
+
+package traps
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/soniah/gosnmp"
+)
+
+// TrapListenerConfig contains configuration for an SNMP trap listener.
+// YAML field tags provided for test marshalling purposes.
+type TrapListenerConfig struct {
+	Version          uint16   `mapstructure:"version" yaml:"version"`
+	Port             uint16   `mapstructure:"port" yaml:"port"`
+	CommunityStrings []string `mapstructure:"community_strings" yaml:"community_strings"`
+	BindHost         string   `mapstructure:"bind_host" yaml:"bind_host"`
```

Comment:
There's a comment above the `struct` declaration that clarifies that these are needed to facilitate tests (I generate the YAML from structs to make sure the conf parsing works, which `snmp.go` doesn't test).

See: https://github.com/DataDog/datadog-agent/pull/5470/files#diff-2defe81bba4f82bb622e334cdc9a85e6R154-R160

Reply:
I see.

However, a slightly unrelated: maybe you can use text in tests instead of Marshaling objects if that's possible (I didn't had time to review the test part yet but will do soon)

Reply:
Yes, it's possible to use text, but we end up having to essentially re-write a struct-to-YAML serializer (or we have to duplicate the config and the YAML file). This way with the `yaml:` field keeps the tests cleaner.